### ISSUE TITLE
Render nested component

### ISSUE THREAD

**Issue by ramik91** [ractivejs/ractive#3262]
Hi,

I have this code example:
```javascript
const MyComponentA = Ractive.extend({
    template: '<h1>Hello from my custom component A</h1>'
});

const MyComponentB = Ractive.extend({
    template: '<h1>Hello from my custom component B</h1>'
});

const InnerComponent = Ractive.extend({
    template: '<div class="custom">{{yield}}</div>'
});

const OuterComponent = Ractive.extend({
    components: {
        InnerComponent
    },
    template: '<div class="outer">{{>content}}</div>'
});

const app = new Ractive({
    components: {
        MyComponentA,
        MyComponentB,
        OuterComponent
    },
    el: document.body,
    template: `<div class="app">
        <OuterComponent>
                <InnerComponent>
                    <MyComponentA/>
                </InnerComponent>
                <InnerComponent>
                    <MyComponentB/>
                </InnerComponent>
        </OuterComponent>
    </div>`
});
```

I know that this code won't render **MyComponentA** neither **MyComponentB** as ```{{yield}}``` keyword actually sets the context to **OuterComponent** which doesn't know how to resolve them.

Is there a way to move ```{{yield}}``` context multiple levels up or am I just doing the whole thing terribly wrong?

Thanks!

**Comment by ceremcem** [comment]
What is the expected result here? Is that what you want: [Playground](https://ractive.js.org/playground/#N4IgLiBcIgNCBnKIBKBDAxmAlgNwKYB0GA9gLYAOJAdvtWAoQLICeAwuVbfQIIAEAXj7oseIvgAeYOgBMAFMAA61Pqr7TKAGzTTIfAOQAeABYBGAHwAJfJs0k+AMwBO5PmRZ8MAVwRhXpSho6MD4eQwB6M3N9ZQBfAEoAbmVlERwCYk4g+kZWDkDuMAAhQWFMdPEpWQVlNXV8LR18PSMo61t7Z1d3Tx8-Mk8swr4iiKiY6gTk6mVSal8+AElqWid8rmDStLFCSWlqeSUVNQ0KbV0DQxk8T20EBAFFEG9fcifzYGAWbBsZWNiItdcNE4kkUtQ5gsAPJeaRrIabITbDJ7apHOoBDY5PToupqZardbZMC1NSxWCk1Snc7NS5A25oe6PEAkWH4JzvT7mOb7MD-QF4EGTMEzCE0BZoCgUUq0ADuZVEBBqx1UmOJCBxlLxMLhRMKWvJWpsehkJG8ZGChAARiQZCwKSr6o0LgADK43DB3B5PSUUd5auqGHXsvXBcwBvGRwwEkMI+jhx2RpNqQx5ONgHjhBPJ5MRGPwgphiM5vjRlaxwvx4sllNpyvFLPV3PhfOhquJlPhYMFrFgbOdoHmF2g6ZwZ7IMfGKCgJzIGm+ED-IA)

**Comment by ramik91** [comment]
@ceremcem Yeah exactly, but i wonder if its possible without using global components registry.

**Comment by ceremcem** [comment]
You mean "fallback mechanism" that a component will search for requested components in the current registry and then will start searching from one level up till global registry (`Ractive.components`) . 

I'm not sure if that would be a desired behavior or it is suitable from the point of view of performance, but I find no objection right now. I'm sure @evs-chris would shed the light on the case.

**Comment by ghost** [comment]
The simplest answer would be to just [use `{{ yield }}` on `OuterComponent`](https://ractive.js.org/playground/#N4IgLiBcIgNCBnKIDGB7AdgsACAsgJ4DCaAtgA6YCmGYAgjgLw4BKAhimAJYBuVAdFQAeYGgBMAFMAA6GHPJyiKAGzajIOAOQAeABYBGAHwAJKsuVocAMwBOZHKQI4UAV2z30FarRx1tAegNDTVkAXwBKAG5ZWXQsXEISLwwaMAAhJlYObj5BEXEpWQVFKhU1Kg0dINNzS1t7R2c3MA8yShSfNICgkIwI6IxYzGwcAEkMFJsk9tTM9k5eAWFRDEkZOQUlclV1LW0xXmdVBARGaVRmsnPDYGACLjMxUNCAg55gsKiYjDiRgHkXKIpm1vLhmPMckt8qtCht5FsdhU9m8jmwTmcQGhAVQbNdbvdHs9XrwPn0voMfsNcGxyORMikAO5ZBZ8WHFTwzWgIDTrYrFAFA6ag2BFPnycaTIUdMAiuEKRIg6V0WVihXJVJpUXyUIqhRmDRiNCuUipfgAIzQYgIuvhpW25Q0AAN9ocUMdTucaeRrlq+doBTipalDL6xWLtBLA4rg6Gw3HtGrOfR-CG5XHigFI8D1bRU+mwxGJlGc2A8-n0wniNHaGkU7H85mi9mk2Xw-4A83Qa2cMT3o7PgM4KhkEPdFBQDZkIjsCBnkA) if you really want `app` to define what's possible on the template. I don't see why you'd need `{{> content }}` on it given this small example.

**Comment by ramik91** [comment]
@fskreuz that works just fine, but in that case **InnerComponent** has to be registered in **app** as well. Thats the way I use it in my project atm.

**Comment by dagnelies** [comment]
Hi,

I know I'm late but I thought I'd add my two cents. Basically, the example does not work because `MyComponentA` and `MyComponentB` are not declared at the right place. Since `OuterComponent` uses `{{>content}}`, it also has to know how to render that content. This is where the A B thingies have to be declared, like this:

```
const OuterComponent = Ractive.extend({
    components: {
        MyComponentA,
        MyComponentB,
        InnerComponent
    },
    template: '<div class="outer">{{>content}}</div>'
});
```
That's it.
